ConsoleBank/Currencies/ICurrenciesApi.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Threading.Tasks;
+using Currencies.Entities;
+
+namespace Currencies
+{
+    public interface ICurrenciesApi
+    {
+        Task<Currency[]> GetCurrencies();
+        Task<CurrencyRate> GetCurrencyRate(int currencyId);
+        Task<CurrencyRate> GetCurrencyRate(string currencyAbbreviation);
+        Task<CurrencyRate> GetCurrencyRate(int currencyId, DateTime ondate);
+        Task<CurrencyRate> GetCurrencyRate(string currencyAbbreviation, DateTime ondate);
```

Comment:
чтобы не плодить такие дубликаты, можно параметры сделать опциональными
```suggestion
        Task<CurrencyRate> GetCurrencyRate(int currencyId, DateTime? ondate);
        Task<CurrencyRate> GetCurrencyRate(string currencyAbbreviation, DateTime? ondate);
```